adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralRTBInterstitialAdLoader.m

```diff
@@ -0,0 +1,122 @@
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationInterstitialAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(ad, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+    
+    NSString *adUnitId = adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+    if (!adUnitId.length || !placementId.length) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Ad Unit ID or Placement ID cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+
+    _interstitialAd = [[MTGNewInterstitialBidAdManager alloc]initWithPlacementId:placementId unitId:adUnitId delegate:self];
+    [_interstitialAd loadAdWithBidToken:adConfiguration.bidResponse];
+}
+
+#pragma mark - MTGNewInterstitialBidAdDelegate
+- (void)newInterstitialBidAdResourceLoadSuccess:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    if (_adLoadCompletionHandler) {
+        _adEventDelegate = _adLoadCompletionHandler(self, nil);
+    }
+}
+
+- (void)newInterstitialBidAdLoadFail:(nonnull NSError *)error adManager:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    if (_adLoadCompletionHandler) {
+        _adLoadCompletionHandler(nil, error);
+    }
+}
+
+- (void)newInterstitialBidAdShowSuccess:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    [_adEventDelegate willPresentFullScreenView];
+    [_adEventDelegate reportImpression];
+}
+
+- (void)newInterstitialBidAdShowFail:(nonnull NSError *)error adManager:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    [_adEventDelegate didFailToPresentWithError:error];
+}
+
+- (void)newInterstitialBidAdClicked:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    [_adEventDelegate reportClick];
+}
+
+- (void)newInterstitialBidAdDismissedWithConverted:(BOOL)converted adManager:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    [_adEventDelegate willDismissFullScreenView];
+}
+
+- (void)newInterstitialBidAdDidClosed:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    [_adEventDelegate didDismissFullScreenView];
+}
+
+#pragma mark - GADMediationInterstitialAd
+- (void)presentFromViewController:(nonnull UIViewController *)viewController {
+    
+    GADMAdapterMintegralExtras *extras = _adConfiguration.extras;
+    _interstitialAd.playVideoMute = extras.muteVideoAudio;
+    if ([_interstitialAd isAdReady]) {
+        [_interstitialAd showFromViewController:viewController];
+    }else{
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorAdFailedToShow, @"Mintegral SDK failed to present an interstitial ad.");
```

Comment:
Nit, "a bidding interstitial ad".


Can you make the same change for other ad formats?

And for waterfall, it should be: "a waterfall <format> ad".

Reply:
ok,I have modified.